src/extensions/default/Inline3DParametersEditor/main.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * File based on InlineColorEditor/main.js
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var EditorManager           = brackets.getModule("editor/EditorManager"),
+        ExtensionUtils          = brackets.getModule("utils/ExtensionUtils"),
+        HTMLUtils               = brackets.getModule("language/HTMLUtils"),
+        Inline3dParametersUtils = require("Parameters3DUtils"),
+        Inline3DParameterEditor = require("Inline3DParameterEditor").Inline3DParameterEditor;
+
+    /*
+	 * Used for determining if the current tag is of type <a-*
+	 * Right now the inline Widget functionality if limited to
+	 * Aframe HTML tags.
+     * @param {tag} tag of the html line containing pos
+     * @return {boolean}
+	 */
+    function is3DParameter(tag) {
+        var tagStart =Inline3dParametersUtils.TAG_START;
+        if(!tag.substr(0, 2) === tagStart) {
+            return false;
+        }
+        return true;
+    }
+
+    /*
+	 * Check if a mathc exists for the PARAMETERS_3D_REGEX
+	 * that contains the position pos.
+     * @param {Editor} hostEditor
+     * @param {{line:Number, ch:Number}} pos
+     * @return {?{parameter:String, marker:TextMarker}}
+	 */
+    function getMatch(hostEditor, pos, tagInfo) {
+        var match, cursorLine, ParameterRegex, start, end, endPos, marker ;
+        ParameterRegex = Inline3dParametersUtils.PARAMETERS_3D_REGEX;
+        ParameterRegex.lastIndex = 0;
+        cursorLine = hostEditor.document.getLine(pos.line);
+        // Loop through each match of ParameterRegex and stop when the one that contains pos is found.
```

Comment:
This comment says what the code does - that is not a good code comment. The code comments should say what the intention of the code is, with the code itself being clear on how it does that. So: what are we doing here? Are we trying to find the start of a parameter sequence? If so, there might be much easier code to achieve this